packages/flutter_tools/lib/src/android/adb.dart

```diff
@@ -112,6 +118,26 @@ class Adb {
     return controller.stream;
   }
 
+  Future _populateDeviceNames(List<AdbDevice> devices) async {
+    for (AdbDevice device in devices) {
+      if (device.modelID == null) {
+        // If we don't have a name of a device in our cache, call `device -l` to populate it.
+        if (_idToNameCache[device.id] == null)
+          await _populateDeviceCache();
+
```

Comment:
It's not critical if we don't get a name back from adb. The code will fall back on generating a name from 'deviceType deviceId' - which doesn't look great, but works.